ddln_common.proto

```diff
@@ -17,6 +17,12 @@ message Quaternion {
     required double z = 4;
 }
 
+message LatLongAlt {
```

Comment:
I don't think we actually need this: on the opposite, we should be able to supply the GPS origin instead of asking for it
what is the usecase for requesting LatLongAlt from simulated GPS?

Reply:
To put this another way: starting GPS position is a mission concern, it should be set in the mission. This should not be part of the flight controller startup environment.

Reply:
We need to know the lat long coordinates of the drone to convert NED coordinates into mission LatLongAlt coordinates. I would say this is a basic service which a sim node should be able to tell us. There could be more use cases for this in the future.

Reply:
These ground truth outputs are all part of the sim node state, it is better not to do double book keeping and ask such info directly from the source.

Reply:
Yes what you say is true but it is another issue / task and independent of getting the current global position.

Reply:
It is totally fine to have temporary solutions inside private repos, but I'd prefer to have final solutions in the public external interface. This is the most immutable thing we have, or at least it should be this way.

You can and should obtain Lat/Lon origin coordinates from Imaginator start point: the start location (or locations) is part of the scene, and should be taken from there. This will also allow us to run the same mission on different scenes without further modifications.

Lat/lon are not logically part of the ground truth. We can have flight controller that doesn't have GPS at all, and its simulation also lacking any concept of Lat/Lon.

Please remove it.

Reply:
Sorry but this is a blocker for my progress. Nobody is talking about temporary solution here. 

1. Can we agree that global position is a state?
2. If so who would be owner of that state?

I would say the simulation node (could be gazebo, airsim, or flight gear) is owner of this state, therefore it should be provided by such. This is unrelated to GPS which is just a sensor.

Reply:
No, it is not.

Just to give you an example: Volocopter has an attitude controller. It probably has some simulation environment they use to develop it, but there's no concept of global position for the attitude controller - so it will probably not be simulated. It will have no concept of latitude or longitude in its code at all.

Global position (lat/lon) only has meaning for sensors that are able to measure it. We do need to coordinate these sensors, so they provide compatible values - we should initialize them to the same starting point. But afterwards there's no need to shuffle lat/lon together with ENU all the time - it is just duplicated information.

Lat/lon is not part of the state - this is a piece of information derived from ENU pose and origin lat/lon. This piece of derived information is recalculated whenever the need arises, it doesn't have to be communicated or stored.

Reply:
Or you can also think of it in another way, lat long alt (or ECEF) is a state and ENU pose is derived from this (using arbitrary home position). Professional simulators are usually described in earth coordinates and have a round earth model. Your logic breaks down for large distances and even the IMU is dependent on the global position as the earth gravitational field varies with latitude and altitude. We are aiming for high and market which implies simulations modeled around the WGS84 earth model. I would say any self respecting manufacturer of commercial aircraft is working with WGS84 based simulation which includes global position as a state. For example in my previous company we did not even have a notion of local position (ENU or NED).

Reply:
Nothing breaks. Put your ENU origin at the center of the earth or some equatorial position and simulate to your hearts desire.

You are right that ENU and lat/lon contain duplicated information. So a viable option is to communicate lat/lon instead of ENU. But there are a few reasons I think this is less convenient:
- it is less readable (ENU raw values are easier to read to understand aircraft behavior)
- some FC/sims may have no concept of lat/lon, but all will have concept of XYZ
- it is easier to move the whole scenario to a different location: for FC simulator literally nothing changes, all communicated poses remain comparable between runs

We can argue weather to communicate XYZ or lat/lon over this interface, but there's no case to communicate both.

Reply:
Regarding your experience at Pilatus: I strongly suspect that the physical simuation was running in some cartesian space (XYZ) and not in lat/lon.

Reply:
We used ECEF coordinate system which I mentioned above. It is trivial to obtain lat long alt from ECEF. ENU is not well defined without also supplying home position, which should be added to the ground truth in this case. Anyway please give it some thought and we discuss Monday morning. Have a nice weekend.

Reply:
> Nothing breaks. Put your ENU origin at the center of the earth or some equatorial position and simulate to your hearts desire.

This is not how ENU is defined. What you describe here is more like ECEF.

Reply:
I though about it and agree with your proposed solution. I've update the pull request.